src/deroff.rs

```diff
@@ -360,16 +360,33 @@ impl Deroffer {
         })
     }
 
-    fn skip_char<'a>(&self, s: &'a str, amount: Option<usize>) -> &'a str {
+    fn skip_char(&mut self, amount: usize) {
+        self.s = self.s.get(amount..).unwrap_or("").to_owned();
```

Comment:
I think it should not overwrite the original bytes.

Reply:
Looking at python source code, this looks like what they are doing. I think it should be good then.

Reply:
Yeah I think (based on the python) it should just move the string forward `amount` chars,
here's the source for reference:
```python
def skip_char(self, amt=1):
    self.s = self.s[amt:]
```

Reply:
Why not?
```suggestion
        self.s.drain(..amount);
```

Reply:
I didn't know `drain` existed on `String` lol. I thought it was only on `Iterator`s but that's definitely clearer.